independent-projects/arc/runtime/src/main/java/io/quarkus/arc/impl/ActivateRequestContextInterceptor.java

```diff
@@ -15,17 +16,46 @@ public class ActivateRequestContextInterceptor {
 
     @AroundInvoke
     Object aroundInvoke(InvocationContext ctx) throws Exception {
+        if (ctx.getMethod().getReturnType().equals(Uni.class)) {
```

Comment:
@mkouba , @geoand do we have a standard way to define if a method is reactive or not? Possibly an existing method to invoke here for consistency?

Reply:
It's also a question of which type we want to support. Should we do the same for CompletionStage? Multi?

Reply:
We don't really have something, but it would likely make sense to add it

Reply:
> It's also a question of which type we want to support. Should we do the same for CompletionStage? Multi?

I'd prefer adding them.

> We don't really have something, but it would likely make sense to add it

right. However this is Arc - not sure if we want that within Arc or if other components would use such a method from Arc.
I guess @DavideD you could extract a static method in Arc, then we see?

Reply:
We have something for translating reactive types from one to another. Once again, just copy or abstract the stuff from `TransactionalInterceptorBase`, it's called `io.smallrye.reactive.converters.ReactiveTypeConverter`.

Reply:
I have a generalized variant of SmallRye Reactive Converters that can also handle Kotlin coroutines in SmallRye Fault Tolerance: https://github.com/smallrye/smallrye-fault-tolerance/tree/main/implementation/core/src/main/java/io/smallrye/faulttolerance/core/invocation It's not exactly super-pretty, but works well :-)